app/src/lib/rules/case_frame/sense_selection.js

```diff
@@ -101,12 +142,22 @@ const verb_sense_rules = [
 		['make-C', { 'patient': { 'stem': 'command|fire|god|peace|promise|wave|covenant' } }],
 		['make-E', { 'patient': { 'stem': 'bread|food' } }],
 	]],
+	['pray', [
+		['pray-D', { }],	// prioritize pray-D over pray-A
+	]],
+	['prepare', [
+		['prepare-B', { 'patient': { 'stem': 'feast|food|meal' } }],
+		['prepare-C', { }],
+	]],
+	['return', [
+		['return-D', { 'destination': { 'stem': 'king|person' } }],
+	]],
 	['say', [
 		['say-D', { 'agent': { 'stem': 'law' } }],
 	]],
 	['see', [
 		['see-D', { 'patient': { 'stem': 'dream|vision' } }],
-		['see-C', { }],	// prioritize see-C over see-B gets selected
+		['see-C', { }],	// prioritize see-C over see-B
```

Comment:
I wonder if we should consider a property to words?  Where the default is lower like 0 giving us the ability to scale that priority up if/when needed?  Right now, it seems it might be positionally based...

Reply:
Yes, it is intentionally positionally based. The comment at the top of this file says:
```
 * These rules help decide which sense to select from the ones that match the argument structure.
 * They are ordered by priority, and can provide additional filtering of the verb or arguments to
 * narrow down the match.
 * A sense can also have multiple rows in case there are multiple incompatible filters to check.```

One problem with assigning a priority number is that the priority may depend on the arguments present and their filters. For example, in the case of 'see', the presence of a Patient argument could mean see-A or see-D. We want to select see-A by default (since it has a higher letter), UNLESS the patient is 'dream' or 'vision', in which case see-D is prioritized over see-A.

I can definitely make the comments/documentation clearer.

Reply:
nah, you can leave it the way it is now.  Hopefully, future maintainers will see that note.